packages/browser/src/plugins/ajs-destination/index.ts

```diff
@@ -139,22 +140,20 @@ export class LegacyDestination implements DestinationPlugin {
       analyticsInstance
     )
 
-    this.onReady = new Promise((resolve) => {
-      const onReadyFn = (): void => {
-        this._ready = true
-        resolve(true)
-      }
+    setTimeout(() => {
+      const e = 'Destination timed out'
+      this.initializePromise.reject(e)
+      this.readyPromise.reject(e)
```

Comment:
Can we add a metric here?

Reply:
I'm also curious about what happens in JS when an already resolved promise rejects. it could happen here that `initializedPromise` was already resolved then `readyPromise` times out and makes everything reject.

Reply:
Should this set `_ready` and `_initialized` to `false`?

Reply:
> I'm also curious about what happens in JS when an already resolved promise rejects

It's a no-op in JS. A settled promise will stay in the same state, further `resolve` or `rejects` don't do anything.

> Can we add a metric here?

Done in the other patch